webapp/src/main/java/com/box/l10n/mojito/rest/commit/CommitWS.java

```diff
@@ -98,17 +100,72 @@ public Page<Commit> getCommits(@RequestParam(value = "repositoryId") Long reposi
     /**
      * Gets the last known commit that we have processed and recorded a
      * PushRun for from the list of commit names provided.
+     * <p>
      * Returns null if no commit is found for that repository with a
      * corresponding push run.
      *
-     * @param lastPushedCommitBody The commit names to search for together with the repositoryID.
+     * @param commitListWithRepositoryIdBody The commit names to search for together with the repositoryID.
      * @return {@link View.Commit}
      */
     @JsonView(View.Commit.class)
     @RequestMapping(value = "/api/commits/lastPushed/", method = RequestMethod.POST)
-    public Commit getLastPushedCommit(@RequestBody LastPushedCommitBody lastPushedCommitBody) {
-        return commitService.getLastPushedCommit(lastPushedCommitBody.getCommitNames(),
-                                                 lastPushedCommitBody.getRepositoryId())
+    public Commit getLastPushedCommit(@RequestBody CommitListWithRepositoryIdBody commitListWithRepositoryIdBody) {
+        return commitService.getLastPushedCommit(commitListWithRepositoryIdBody.getCommitNames(),
+                                                 commitListWithRepositoryIdBody.getRepositoryId())
+                .orElse(null);
+    }
+
+    /**
+     * Gets the last known PushRun that we have processed and recorded against a
+     * commit from the list of commit names provided.
+     * <p>
+     * Returns null if no commit is found for that repository with a
+     * corresponding push run.
+     *
+     * @param commitListWithRepositoryIdBody The commit names to search for together with the repositoryID.
+     * @return {@link View.Commit}
+     */
+    @JsonView(View.CommitDetailed.class)
+    @RequestMapping(value = "/api/commits/lastPushRun/", method = RequestMethod.POST)
+    public PushRun getLastPushRun(@RequestBody CommitListWithRepositoryIdBody commitListWithRepositoryIdBody) {
+        return commitService.getLastPushRun(commitListWithRepositoryIdBody.getCommitNames(),
+                                                 commitListWithRepositoryIdBody.getRepositoryId())
+                .orElse(null);
```

Comment:
Spring doesn't handle optional out of the box?

Reply:
We never return Optional<T> in our API signatures - planning to stick with the same for consistency and not mixing return types.